test/integration/execute.integration.js

```diff
@@ -31,5 +31,17 @@ describe('execute', function() {
         done();
       });
     });
+
+  it('accepts 4 arguments to match defn in the loopback-datasource-juggler',
+    function(done) {
+      const ds = createDataSource();
+      ds.connector.execute('INFO', ['keyspace'], {}, function(err, result) {
```

Comment:
Considering that you are interested in fixing `execute` when invoked via a DataSource, I think it would be better to call `ds.execute` instead of `ds.connector.execute` in the test. Thoughts?

```ts
it('...', async () => {
  const result = await ds.execute('INFO', ['keyspace'], {});
  expect(result).to.be.instanceOf(Buffer);
  // ...
});
```

Reply:
Thanks @bajtos I made this change as well.